fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/KeySpacePath.java

```diff
@@ -198,6 +198,18 @@ default Tuple toTuple(@Nonnull FDBRecordContext context) {
     @Nonnull
     CompletableFuture<ResolvedKeySpacePath> toResolvedPathAsync(@Nonnull FDBRecordContext context);
 
+    /**
+     * Resolves the given key within this path.
+     * @param context the transaction to lookup any necessary directory layer entries.
+     * @param key a raw key from the database
+     * @return the {@link ResolvedKeySpacePath} corresponding to that key, with a potential remainder.
+     */
+    @API(API.Status.EXPERIMENTAL)
+    @Nonnull
+    default CompletableFuture<ResolvedKeySpacePath> toResolvedPathAsync(@Nonnull FDBRecordContext context, byte[] key) {
```

Comment:
Is this doing work similar to `keySpace.resolveFromKey`? What is the difference? Is it because of efficiency (not needing to parse the key from the start every time)?

Reply:
It's primarily here because we want to export from a `KeySpacePath`, and that does not have a reference back to the `KeySpace`. The majority of the logic is shared.
It probably has some efficiency gains if this path is on one of the last sub-directories, but it is still resolving this `KeySpacePath` on every call.